core/modules/tiddler.js

```diff
@@ -12,6 +12,45 @@ Extension methods for the $tw.Tiddler object (constructor and methods required a
 /*global $tw: false */
 "use strict";
 
+/*
+Returns names of tiddler fields.
+
+default: the current tiddler's custom fields, or via...
+  => getFieldNames("$custom")
+all fields:
+  => getFieldNames("$all")	
+standard fields:
+  => getFieldNames("$standard")
+standard text fields:
+  => getFieldNames("$text")
+*/
+
+exports.getFieldNames = function(which) {
```

Comment:
The layout of the function is a little confusing. We've got $text and $standard appearing in one place, $all somewhere else, and $custom doesn't even appear in the code explicitly. It would be good if it was clearer and easier to see how each possible value for "which" was being interpreted. What's wrong with a straightforward switch statement that lists all the options explicitly?